### ISSUE TITLE
Consider the licenses of upstream data source

### ISSUE THREAD

**Issue by dhimmel** [OpenBioLink/OpenBioLink#5]
Looking at the [supplemental tables](https://github.com/OpenBioLink/OpenBioLink/blob/66975836859b6a3b6dc061e2003db53829d438e2/paper/supplementary%20data.pdf), I see the study has incorporated data from many publicly available academic resources and released a derivative integrated dataset. This is a great thing to do, but requires some legal judgement and research. From my experience:

- [Data copyright and licensing section of Rephetio manuscript](https://git.dhimmel.com/rephetio-manuscript/#data-copyright-and-licensing)
- [A discussion we had with some lawyers and librarians](https://doi.org/10.15363/thinklab.d107)
- [news article](https://www.nature.com/news/legal-confusion-threatens-to-slow-data-science-1.20359)

Here is a collection of resources that went into the OpenBioLink network and their licenses:

1. **CC0**: DO, Bgee, Reactome
2. **CC BY**: GO, STRING, Uberon as per https://github.com/obophenotype/uberon/issues/1139, possibly STITCH (depends on [which files](http://stitch.embl.de/cgi/download.pl?UserId=5Fj3EEoFbXhI&sessionId=kRxaXQ2mnvj0))
3. **CC BY SA**: DrugCentral
4. **CC BY-NC-SA**: [SIDER](http://sideeffects.embl.de/)
5. **Custom licenses**: [Comparative Toxicogenomics Database](http://ctdbase.org/about/legal.jsp), [Human Phenotype Ontology](https://hpo.jax.org/app/license), and [KEGG](https://www.kegg.jp/kegg/legal.html)
6. **Unclear**: PubChem (government work not subject to copyright, but PubChem may include third-party content).

CC0, CC BY, CC BY-SA are all open licenses. If you remix content with these licenses, you are required to release your outputs under CC BY-SA due to its copyleft provision.

CC BY-NC-SA is incompatible with CC BY-SA. The custom licenses are iffy. KEGG is most explicit that reusing its data is forbidden without a special agreement.

In all cases, you can get permission from the creators to reuse the data. For things with an open license, you can just reuse them under the license. You can also decide that your reuse falls under "fair use" (a jurisdiction dependent concept) or that it is not subject to copyright since it is purely factual. These arguments are stronger in the U.S. than in Austria I believe.

So my suggestion is to look into the licensing and decide what to do. At a minimum, you should update the readme and manuscript to make readers aware of potential issues.

As a benchmark dataset, I think releasing the network under an open license is important (currently I don't see a license in the repo). However, that will be hard given some of the upstream licenses. With Hetionet, we released our contributions under CC0 and warned users of the complex data licensing / copyright situation.

**Comment by matthias-samwald** [comment]
We added the following to the README:

# Source databases and their licenses

| source type                    | source name                                  | license                                     |
|--------------------------------|----------------------------------------------|---------------------------------------------|
| edge (gene-gene)               | [STRING](https://string-db.org/)             | CC BY                                       |
| edge (gene-go)                 | [GO](http://geneontology.org/)               | CC BY                                       |
| edge (gene-disease)            | [DisGeNet](https://www.disgenet.org/)        | CC BY-NC-CA                                 |
| edge (gene-phenotype)          | [HPO](https://hpo.jax.org/app/)              | Custom: [HPO](https://hpo.jax.org/app/license)      |
| edge (gene-anatomy)            | [Bgee](https://bgee.org/)                    | CC 0                                        |
| edge (gene-drug)               | [STITCH](http://stitch.embl.de/)             | CC BY                                       |
| edge (gene-pathway)            | [CTD](http://ctdbase.org/)                   | Custom: [CTD](http://ctdbase.org/about/legal.jsp)   |
| edge (disease-phenotype)       | [HPO](https://hpo.jax.org/app/)              | Custom: [HPO](https://hpo.jax.org/app/license)      |
| edge (disease-drug)            | [DrugCentral](http://drugcentral.org/)       | CC BY-SA                                    |
| edge (drug-phenotype)          | [SIDER](http://sideeffects.embl.de/)         | CC BY-NC-CA                                 |
| ontology (genes)               | [GO](http://geneontology.org/)               | CC BY                                       |
| ontology (diseases)            | [DO](http://disease-ontology.org/)           | CC 0                                        |
| ontology (phenotype)           | [HPO](https://hpo.jax.org/app/)              | Custom: [HPO](https://hpo.jax.org/app/license)      |
| ontology (anatomy)             | [UBERON](http://uberon.github.io/about.html) | CC BY                                       |
| mapping (UMLS-DO)              | [DisGeNet](https://www.disgenet.org/)        | CC BY-NC-CA                                 |
| mapping (STRING-NCBI)          | [STRING](https://string-db.org/)             | CC BY                                       |
| mapping (ENSEMBL/UNIPROT-NCBI) | [UniProt](https://www.uniprot.org/)          | CC BY                                       |
| id (genes)                     | [NCBI](https://www.ncbi.nlm.nih.gov/gene)    | Public Domain                               |
| id (go)                        | [GO](http://geneontology.org/)               | CC BY                                       |
| id (anatomy)                   | [UBERON](http://uberon.github.io/about.html) | CC BY                                       |
| id (disease)                   | [DO](http://disease-ontology.org/)           | CC 0                                        |
| id (drug)                      | [PubChem](https://pubchem.ncbi.nlm.nih.gov/) | Public Domain                               |
| id (phenotype)                 | [HPO](https://hpo.jax.org/app/)              | Custom: [HPO](https://hpo.jax.org/app/license)      |
| id (pathway)                   | [REACTOME](https://reactome.org/)            | CC BY                                       |
| id (pathway)                   | [KEGG](https://www.genome.jp/kegg/)          | Custom: [KEGG](https://www.kegg.jp/kegg/legal.html) |

The OpenBioLink benchmark files integrate data or identifiers from these sources. The provenance of data items is captured in the benchmark files, and licensing terms of source databases apply to these data items. Please mind these licensing terms when utilizing or redistributing the benchmark files or derivatives thereof.

All original data in the benchmark files created by the OpenBioLink project (not covered by the licenses of external data sources)  are released as [CC 0](https://creativecommons.org/publicdomain/zero/1.0/). 

We offer the benchmark files as-is and make no representations or warranties of any kind concerning the benchmark files, express, implied, statutory or otherwise, including without limitation warranties of title, merchantability, fitness for a particular purpose, non infringement, or the absence of latent or other defects, accuracy, or the present or absence of errors, whether or not discoverable, all to the greatest extent permissible under applicable law.

**Comment by matthias-samwald** [comment]
@annabreit Regarding KEGG: We are not actually redistributing KEGG data, only using KEGG identifiers / mappings to KEGG identifiers, correct?

**Comment by annabreit** [comment]
Imo, yes. We are integrating data from Comparative Toxicogenomics Database, which (also) uses the identifiers of KEGG for their pathway-gene interactions. CTD actually also integrates the data from KEGG, however we do not use/integrate pathway data.

**Comment by matthias-samwald** [comment]
→ We have adopted the approach taken by HetioNet, i.e., licensing any original contributions to the dataset by our project as CC0 while maintaining the licensing restrictions of data items from external datasets. We have extended the GitHub ReadMe with a table detailing the licensing terms of integrated datasets and and a reminder that they should be taken into consideration when redistributing the benchmark or any derivatives. 

We have added information on the source of each data item (i.e., relation) to the benchmark files. (soon to be uploaded)

We added the following sentence to the manuscript: “Licensing terms of integrated datasets are detailed in the documentation, and should be taken into consideration when redistributing the benchmark or any derivative work.”
